src/hooks/use-wp-list.ts

```diff
@@ -0,0 +1,60 @@
+import * as Sentry from '@sentry/electron/renderer';
+import { useState, useEffect } from 'react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export type WpListType = 'plugins' | 'themes';
+export type WpListEntry = { name: string; type: 'file' | 'folder' };
+export type WpListResult = { items: WpListEntry[]; isLoading: boolean; error: Error | null };
+
+export function useWpList(
+	siteId: string,
+	types: WpListType[]
+): Record< WpListType, WpListResult > {
+	const [ results, setResults ] = useState< Record< WpListType, WpListResult > >( () => {
+		const initialResults = {
+			plugins: { items: [], isLoading: true, error: null },
+			themes: { items: [], isLoading: true, error: null },
+		} as Record< WpListType, WpListResult >;
```

Comment:
Let's avoid casting types when possible:

```suggestion
		const initialResults: Record< WpListType, WpListResult > = {
			plugins: { items: [], isLoading: true, error: null },
			themes: { items: [], isLoading: true, error: null },
		};
```